ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/BlockRangeBroadcaster.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright contributors to Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.eth.sync;
+
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.eth.manager.EthContext;
+import org.hyperledger.besu.ethereum.eth.manager.EthMessage;
+import org.hyperledger.besu.ethereum.eth.manager.EthPeer;
+import org.hyperledger.besu.ethereum.eth.messages.BlockRangeUpdateMessage;
+import org.hyperledger.besu.ethereum.eth.messages.EthProtocolMessages;
+import org.hyperledger.besu.ethereum.p2p.rlpx.connections.PeerConnection;
+import org.hyperledger.besu.ethereum.p2p.rlpx.wire.messages.DisconnectMessage;
+import org.hyperledger.besu.ethereum.rlp.RLPException;
+
+import java.time.Duration;
+import java.util.List;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BlockRangeBroadcaster {
+  private static final Logger LOG = LoggerFactory.getLogger(BlockRangeBroadcaster.class);
+
+  // range update block interval
+  static final int BLOCK_RANGE_UPDATE_INTERVAL = 60;
+
+  private final EthContext ethContext;
+  private final Blockchain blockchain;
+
+  public BlockRangeBroadcaster(final EthContext ethContext, final Blockchain blockchain) {
+    this.ethContext = ethContext;
+    this.blockchain = blockchain;
+    ethContext
+        .getEthMessages()
+        .subscribe(EthProtocolMessages.BLOCK_RANGE_UPDATE, this::handleBlockRangeUpdateMessage);
+    ethContext
+        .getScheduler()
+        .scheduleFutureTaskWithFixedDelay(
+            this::broadcastBlockRange,
+            Duration.ofSeconds(BLOCK_RANGE_UPDATE_INTERVAL),
+            Duration.ofSeconds(BLOCK_RANGE_UPDATE_INTERVAL));
```

Comment:
has the spec change to broadcast every minute instead of every 32 blocks?

Reply:
Yes, it is up to the node to decide how often it wants to send the update.

https://github.com/ethereum/devp2p/pull/262

The spec mentions every 2 minutes.